lib/internal/shell.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+
+// There is no need to add primordials to this file.
+// `shell.js` is a script only executed when `node run <script>` is called.
+
+const forbiddenCharacters = /[\t\n\r "#$&'()*;<>?\\`|~]/;
+
+/**
+ * Escapes a string to be used as a shell argument.
+ *
+ * Adapted from `promise-spawn` module available under ISC license.
+ * Ref: https://github.com/npm/promise-spawn/blob/16b36410f9b721dbe190141136432a418869734f/lib/escape.js
+ * @param {string} input
+ */
+function escapeShell(input) {
```

Comment:
should we check for `typeof input !== 'string'`?

Reply:
Not necessarily since we are testing the function that calls this.